packages/material-ui/src/styles/useThemeProps.js

```diff
@@ -0,0 +1,21 @@
+import { getThemeProps } from '@material-ui/styles';
+import useTheme from './useTheme';
+import defaultTheme from './defaultTheme';
+
+export default function useThemeProps({ props: inputProps, name }) {
+  /* eslint-disable prefer-object-spread */
+  const props = Object.assign({}, inputProps);
```

Comment:
Why assign and not spread?

Reply:
Will refactor it 👍 I don't remember if there was particular reason for it.